src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Bundle",
+				id = bundleId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	MarketplaceService.PromptGamePassPurchaseFinished:Connect(
+		function(player: Player, gamePassId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "GamePass",
+				id = gamePassId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	if LOG_DEV_PRODUCT_PURCHASES then
+		MarketplaceService.PromptProductPurchaseFinished:Connect(
+			function(player: Player, productId: number, purchased: boolean)
+				if not purchased then
+					return
+				end
+				
+				self:LogMarketplacePurchase({
+					userId = player.UserId,
+					itemType = "Product",
+					id = productId,
+					cartType = "PromptPurchase",
+				})
+			end
+		)
+	end
+	
+	MarketplaceService.PromptPurchaseFinished:Connect(
+		function(player: Player, assetId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Asset",
+				id = assetId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	MarketplaceService.PromptSubscriptionPurchaseFinished:Connect(
+		function(player: Player, subscriptionId: string, didTryPurchasing: boolean)
+			if not didTryPurchasing then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Subscription",
+				id = subscriptionId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+end
+
+--[=[
+	Used to track player events (example: player killed an enemy, player completed a mission, etc.)
+	
+	Examples
+	```lua
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:KilledEnemy",
+		value = 1 -- Killed 1 enemy
+	})
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:CompletedMission",
+		value = 1 -- Completed 1 mission
+	})
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:Death",
+		value = 1
+	})
+	```
+]=]
+function AnalyticsService:LogPlayerEvent(data: PlayerEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogPlayerEvent - data is required")
+	assert(data.userId, "AnalyticsService.LogPlayerEvent - userId is required")
+	assert(data.event, "AnalyticsService.LogPlayerEvent - event is required")
+	assert(
+		data.value == nil or typeof(data.value) == "number",
+		"AnalyticsService.LogPlayerEvent - value must be a number"
+	)
+	
+	GameAnalytics:addDesignEvent(data.userId, {
+		eventId = data.event,
+		value = data.value
+	})
+end
+
+--[=[
+	This function should be called when a successful marketplace purchase is made
+	such as a gamepass or developer product
+	
+	Set `LOG_DEV_PRODUCT_PURCHASES` to false inside AnalyticsService.lua if you prefer to log
```

Comment:
If you define a @prop tag for LOG_DEV_PRODUCT_PURCHASES, you can hyperlink it here.